### ISSUE TITLE
[BUG][v1.5.3] VMAF functions exit with error when vmaf_read_pictures and vmaf_score_pooled calls are interleaved

### ISSUE THREAD

**Issue by funroger** [Netflix/vmaf#712]
Good day,

I'm implementing a video coding system. To make sure, that video encoding is OK, I query VMAF score after every **vmaf_read_pictures** call.
Unfortunately, VMAF library calls return error after 3rd call of **vmaf_read_pictures**. I debugged and found the reasons: both **vmaf_read_pictures** and **vmaf_score_pooled** cause updating a vector with results for VMAF_feature_motion2_score feature.
While function **vmaf_read_pictures** updates result number picture_index - 1 (float_motion.c:165), function **vmaf_score_pooled** calls flush functions and updates result number picture_index (float_motion:95).
Some time later, when I add frame number 2, function **feature_vector_append** returns an error, because the corresponding vector item is already written from previous flush call.

Summing up, libvmaf works until the first call of **vmaf_score_pooled**. Shortly after the library returns error from most of its methods.

Repro code is very simple:
```
int frameIdx = 0;
while (!end_of_stream) {
  // obtain encoded and reference frames

  // provide frames to vmaf library
  res = vmaf_read_pictures(ctx, &ref, &src, frameIdx);
  if (res) {break;}

  // print the current frame score
  res = vmaf_score_pooled(ctx, model, VMAF_POOL_METHOD_MEAN, &score, frameIdx, frameIdx);
  if (res) {break;}

  ++frameIdx;
}

// get the whole stream score (ALWAYS FAIL)
if (frameIdx) {
  res = vmaf_score_pooled(ctx, model, VMAF_POOL_METHOD_MEAN, &score, 0, frameIdx - 1);
  if (res) {return -1;}
}

```

Could you fix it please?
I used official v1.5.3 release. 100% repro on Linux and Windows.

**Comment by kylophone** [comment]
Yes, current VMAF models have a temporal feature called `motion` which requires a future frame. Right now, this feature is only written for frame **n** when frame **n+1** is read. This feature is fortunately reference only, meaning this only pixel data from the reference frame is required. An new API will need to be created to compute reference-only features. This will solve your problem as well as OOO picture coding. It's on the roadmap, but I'll move it towards the top of the list.

**Comment by funroger** [comment]
Thank you! I'll be waiting.

**Comment by funroger** [comment]
Thank you for the quick fix.

**Comment by funroger** [comment]
Hi,

I tested the latest master. Sorry, I can't confirm that the issue is fixed.
Only the first call of **vmaf_score_pooled** returns some VMAF value. All subsequent **vmaf_score_pooled** calls return VMAF score equal 0.0. The last after-the-cycle call **vmaf_score_pooled** returns an error.

I debugged a bit the last call. It returns the error because of error checking in feature_collector.c:300 :
```

    if (!feature_vector->score[index].written) {
        err = -EINVAL;
        goto unlock;
    }

```
feature_vector->score[index].written is false. Looks like a thread racing or something.
Regards,
Victor
